### ISSUE TITLE
IN()  with empty set

### ISSUE THREAD

**Issue by fablue** [speedment/speedment#492]
Using the .in() function and passing an empty set to it returns in a SQLException
```
Error executing SELECT `id`,`visibility`,`token`,`firebase`,`initial`,`timestamp_location`,`distance`,`longitude`,`latitude` FROM `lime`.`user` WHERE ((`lime`.`user`.`firebase` IS NOT NULL) AND (((`lime`.`user`.`latitude` >= ? AND `lime`.`user`.`latitude` < ?) AND (`lime`.`user`.`longitude` >= ? AND `lime`.`user`.`longitude` < ?)) OR (`lime`.`user`.`id` IN ()))) ORDER BY `lime`.`user`.`timestamp_location` DESC, values=[49.0224494934082, 49.02457809448242, 12.060419082641602, 12.062543869018555]
java.sql.SQLSyntaxErrorException: (conn:162) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')))) ORDER BY `lime`.`user`.`timestamp_location` DESC' at line 1
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:158)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:118)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:229)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:208)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.execute(MariaDbPreparedStatementClient.java:147)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeQuery(MariaDbPreparedStatementClient.java:161)
	at com.speedment.runtime.core.internal.db.AsynchronousQueryResultImpl.stream(AsynchronousQueryResultImpl.java:95)
	at com.speedment.runtime.core.internal.component.sql.SqlStreamSupplierImpl.lambda$stream$6(SqlStreamSupplierImpl.java:184)
	at com.speedment.runtime.core.internal.stream.builder.pipeline.PipelineImpl.getStream(PipelineImpl.java:85)
	at com.speedment.runtime.core.internal.stream.builder.pipeline.PipelineImpl.getAsReferenceStream(PipelineImpl.java:63)
	at com.speedment.runtime.core.internal.component.sql.override.def.reference.DefaultCollectTerminator.apply(DefaultCollectTerminator.java:49)
	at com.speedment.runtime.core.internal.manager.sql.SqlStreamTerminator.collect(SqlStreamTerminator.java:141)
	at com.speedment.runtime.core.internal.stream.builder.ReferenceStreamBuilder.lambda$collect$8(ReferenceStreamBuilder.java:289)
	at com.speedment.runtime.core.internal.stream.builder.AbstractStreamBuilder.finallyCloseReference(AbstractStreamBuilder.java:189)
	at com.speedment.runtime.core.internal.stream.builder.ReferenceStreamBuilder.collect(ReferenceStreamBuilder.java:289)
```

Reproducable with:
```java
Set<Long> set = new TreeSet<>();
manager.stream().filter(Entity.ID.in(set)).findAny();
```
And MariaDB (driver 2.0.1)

Therefore I have to handle the case of an empty set everytime which results in not that beautiful code. 
I think the expected behavior would be something like an always "negative" predicate?

**Comment by minborg** [comment]
This is certainly something we have to correct. I have marked it as a bug and will correct it ASAP.

**Comment by Pyknic** [comment]
I have submitted a fix to this and #493 in `develop-3.0` now. In-predicates with no elements are rendered as `(FALSE)`, single elements are rendered as `(column = ?)` and more than one element is rendered as before. It might be more aesthetic  if empty sets prevented any query from being executed at all, but that requires a bit more advanced changes to the query engine.
